### DISCUSSION TITLE
PSBT output address diff with rgb address

### DISCUSSION THREAD

**Discussion by deepseeksss** [RGB-WG/rgb#125] [Q&A]
1) contruct psbt, the address derive is default to keychain 10 and without taproot tweaks

2) rgb address -k 10 -i index, it derived the address with taproot tweaks

the result is step 1 and step 2 address is not same even though the keychain and index are all same

is this behavior normal?
how to sign the psbt?

**Comment by dr-orlovsky** [comment]
> contruct psbt, the address derive is default to keychain 10 and without taproot tweaks

Nope, it is always with a taproot tweak, since we have to put a tweak into the transaction

> rgb address -k 10 -i index, it derived the address with taproot tweaks

No, this one is WITHOUT taproot tweak, since there is no tweak to apply.

> is this behavior normal?
Yes

> how to sign the psbt?
The normal way - it is outputs which are tweaked, so any signer will work. It is just the  hardware wallets will see the change as an external payments - but there is no way around that until we will have a dedicated RGB app for each of hardware wallets

Overall, it is advised not to use `-k` flag, like in `rgb address -k 10 -i index` unless you are clearly known what you are doing.

Please feel free to re-open if you have any further questions

**Reply by deepseeksss** [reply]
@dr-orlovsky  thank you for your reply, i'm sure that `rgb address` command is derive with tweak, because of the tweaks in the cache.yaml is not empty, but when construct the pstb it will not with the tweak, so when i try to sign the psbt it will stucked

**Reply by dr-orlovsky** [reply]
`rgb address` derives without tweaks. There might be multiple tweaks for the same derivation path, it is unclear how to structure the interface around that.
